applications/GeoMechanicsApplication/tests/cpp_tests/test_stress_strain_utitlities.cpp

```diff
@@ -0,0 +1,115 @@
+    KRATOS_EXPECT_DOUBLE_EQ(0.0, eye(1, 2));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculateVonMisesStress, KratosGeoMechanicsFastSuite)
+{
+    Vector stress_vector(4);
+    stress_vector <<= -2.0, -2.0, -2.0, 0.0;
+    KRATOS_EXPECT_DOUBLE_EQ(0.0, StressStrainUtilities::CalculateVonMisesStress(stress_vector));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculateLodeAngle, KratosGeoMechanicsFastSuite)
+{
+    Vector stress_vector(4);
+    stress_vector <<= 0.5, -1.0, 0.5, 0.0;
+    KRATOS_EXPECT_DOUBLE_EQ(MathUtils<>::DegreesToRadians(30.),
+                            StressStrainUtilities::CalculateLodeAngle(stress_vector));
+    KRATOS_EXPECT_DOUBLE_EQ(MathUtils<>::DegreesToRadians(-30.),
+                            StressStrainUtilities::CalculateLodeAngle(-1. * stress_vector));
+    Vector stress_vector2(4);
+    stress_vector2 <<= -1.0, 0.0, 1.0, 0.0;
+    KRATOS_EXPECT_DOUBLE_EQ(MathUtils<>::DegreesToRadians(0.),
+                            StressStrainUtilities::CalculateLodeAngle(stress_vector2));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculateMCShearCapacityZeroStress, KratosGeoMechanicsFastSuite)
+{
+    Vector stress_vector  = ZeroVector(4);
+    double cohesion       = 2.;
+    double friction_angle = 0.;
+    KRATOS_EXPECT_DOUBLE_EQ(
+        0.0, StressStrainUtilities::CalculateMCShearCapacity(stress_vector, cohesion, friction_angle));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculateMCShearCapacityHydrostatic, KratosGeoMechanicsFastSuite)
+{
+    Vector stress_vector(4);
+    stress_vector <<= -2.0, -2.0, -2.0, 0.0;
+    double cohesion       = 0.;
+    double friction_angle = MathUtils<>::DegreesToRadians(90.);
+    KRATOS_EXPECT_DOUBLE_EQ(
+        0.0, StressStrainUtilities::CalculateMCShearCapacity(stress_vector, cohesion, friction_angle));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculateMCShearCapacityShearOnly, KratosGeoMechanicsFastSuite)
+{
+    Vector stress_vector(4);
+    stress_vector <<= -2.0, 0.0, 2.0, 0.0;
+    double cohesion       = 2.;
+    double friction_angle = 0.0;
+    KRATOS_EXPECT_DOUBLE_EQ(
+        1.0, StressStrainUtilities::CalculateMCShearCapacity(stress_vector, cohesion, friction_angle));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculateMCPressureCapacityZeroStress, KratosGeoMechanicsFastSuite)
+{
+    Vector stress_vector  = ZeroVector(4);
+    double cohesion       = 2.;
+    double friction_angle = 0.;
+    KRATOS_EXPECT_DOUBLE_EQ(
+        0.0, StressStrainUtilities::CalculateMCPressureCapacity(stress_vector, cohesion, friction_angle));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculateMCPressureCapacityShearOnly, KratosGeoMechanicsFastSuite)
+{
+    Vector stress_vector(4);
+    stress_vector <<= -2.0, 0.0, 2.0, 0.0;
+    double cohesion       = std::sqrt(2.*2.*3.);
+    double friction_angle = MathUtils<>::DegreesToRadians(30.0);
+    KRATOS_EXPECT_NEAR(
+        3.*std::sin(friction_angle), StressStrainUtilities::CalculateMCPressureCapacity(stress_vector, cohesion, friction_angle), 1.E-10);
+}
```

Comment:
This test code needs to be reformatted using clang-format.

Reply:
Done